mithril-signer/src/single_signer.rs

```diff
@@ -145,11 +146,15 @@ impl SingleSigner for MithrilSingleSigner {
             signers_with_stake,
             &protocol_initializer.get_protocol_parameters().into(),
         )
+        .with_context(|| "Mithril Single Signer can not compute aggregate verification key")
         .map_err(SingleSignerError::AggregateVerificationKeyComputationFailed)?;
```

Comment:
```suggestion
        .map_err(SingleSignerError::AggregateVerificationKeyComputationFailed(anyhow!("Mithril Single Signer can not compute aggregate verification key")))?;
```

`AggregateVerificationKeyComputationFailed` is defined as `AggregateVerificationKeyComputationFailed(StdError)`. Isn't it easier to add context like this?

Reply:
The `map_err` takes a closure as argument: this is the shortcut for `.map_err(|e|  SingleSignerError::AggregateVerificationKeyComputationFailed(e)`. 
Your proposition is creating a different error actually which is not the expected behavior here :slightly_smiling_face: